.github/workflows/future_continuous_integration.yaml

```diff
@@ -0,0 +1,91 @@
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v3
+      with:
+        ref: main
+        fetch-depth: 1
+
+    - name: Check Changed Subdirectories
+      id: change-monitor
+      run: |
+        TARGET_DIRECTORY="python_src/src/lamp_py"; AWK_FIELD=4
+
+        # Get the list of changed subdirectories
+        CHANGED_SUBDIRS=$(git diff --no-commit-id --dirstat=files,0 \
+                          -r origin/main HEAD -- "${TARGET_DIRECTORY}" | \
+                          awk -F'[/\t]' -v field=$AWK_FIELD '{ print $field }' \
+                          | sort -u)
+
+        # Get the list of changed files for toml and lock files
+        CHANGED_FILES=$(git diff --no-commit-id --name-only \
+                        -r origin/main HEAD -- "python_src/pyproject.toml" \
+                        "python_src/poetry.lock")
+
+        # Initialize test booleans
+        TEST_INGESTION='false'
+        TEST_AWS='false'
+        TEST_PM='false'
+        TEST_POSTGRES='false'
+
+        # Iterate through changed subdirectories and set flags to true based on
+        # what subdirs have changed.
+        for SUBDIR in $CHANGED_SUBDIRS; do
+          if (echo "$SUBDIR" | grep -q -E '^(ingestion|postgres)$'); then
+            TEST_INGESTION='true'
+          fi
+          if (echo "$SUBDIR" | grep -q -E '^aws$'); then
+            TEST_AWS='true'; TEST_INGESTION='true'; TEST_PM='true'
+          fi
+          if (echo "$SUBDIR" | grep -q -E '^(performance_manager|postgres)$'); then
+            TEST_PM='true'
+          fi
+          if (echo "$SUBDIR" | grep -q -E '^postgres$'); then
+            TEST_POSTGRES='true'
+          fi
+        done
+
+        # Check if toml and lock files are changed
+        if [ -n "$CHANGED_FILES" ]; then
+          TEST_INGESTION='true'
+          TEST_AWS='true'
+          TEST_PM='true'
+          TEST_POSTGRES='true'
+        fi
+
+        # Set the output variables using environment files
+        echo "TEST_INGESTION=$TEST_INGESTION" >> $GITHUB_ENV
+        echo "TEST_AWS=$TEST_AWS" >> $GITHUB_ENV
+        echo "TEST_PERFORMANCE_MANAGER=$TEST_PM" >> $GITHUB_ENV
+        echo "TEST_POSTGRES=$TEST_POSTGRES" >> $GITHUB_ENV
+
+
+    - name: Print all outputs
+      run: |
+        echo "Test Ingestion: ${{ env.TEST_INGESTION }}"
+        echo "Test AWS: ${{ env.TEST_AWS }}"
+        echo "Test Performance Manager: ${{ env.TEST_PERFORMANCE_MANAGER }}"
+        echo "Test Postgres: ${{ env.TEST_POSTGRES }}"
+
+    - name: Test Ingestion
+      if: env.TEST_INGESTION == 'true'
+      run: echo "Running Ingestion tests..."
+
+    - name: Test AWS
+      if: env.TEST_AWS == 'true'
+      run: echo "Running AWS tests..."
+
+    - name: Test Performance Manager
+      if: env.TEST_PERFORMANCE_MANAGER == 'true'
+      run: echo "Running Performance Manager tests..."
+
+    - name: Test Postgres 
+      if: env.TEST_POSTGRES == 'true'
+      run: echo "Running Postgres tests..."
```

Comment:
the idea is that this section will eventually have `poetry run pytest tests/aws` or whatever. we'll also standup the postgres server only for the tests that require them.

Reply:
I'm not totally sure what all is happening in "Check Changed Subdirectories".

It does look there are some hard coded file names and command parameters that we need to be very aware of, which changes to folder structure in the project. Might be worth adding that information as comments and also maybe some instructions on how to handle the additions of new library folders.